eslint.config.mjs

```diff
@@ -0,0 +1,336 @@
+        ],
+
+        rules: {
+            '@typescript-eslint/no-var-requires': 'off',
+        },
+    },
+    {
+        files: ['packages/lwc/**'],
+
+        rules: {
+            'no-restricted-imports': 'off',
+        },
+    },
+    {
+        files: [
+            'packages/@lwc/engine-core/**',
+            'packages/@lwc/engine-dom/**',
+            'packages/@lwc/synthetic-shadow/**',
+        ],
+
+        rules: {
+            '@lwc/lwc-internal/no-global-node': 'error',
+            'prefer-rest-params': 'off',
+            'prefer-spread': 'off',
+        },
+    },
+    {
+        files: ['**/__tests__/**', '**/__mocks__/**', 'packages/@lwc/integration-karma/**'],
+
+        languageOptions: {
+            globals: {
+                ...globals.jest,
+            },
+        },
+
+        rules: {
+            'jest/no-focused-tests': 'error',
+            'jest/valid-expect': 'error',
+            'jest/valid-expect-in-promise': 'error',
+        },
+    },
+    {
+        files: ['packages/@lwc/integration-tests/**'],
+
+        languageOptions: {
+            globals: {
+                $: true,
+                browser: true,
+            },
+        },
+    },
+    {
+        files: ['./*.js', '**/scripts/**', '**/jest.config.js'],
+
+        languageOptions: {
+            globals: {
+                ...globals.node,
+                ...globals.jest,
+            },
+        },
+
+        rules: {
+            'no-console': 'off',
+        },
+    },
+    {
+        files: ['packages/@lwc/perf-benchmarks/**'],
+
+        languageOptions: {
+            globals: {
+                after: true,
+                before: true,
+                benchmark: true,
+                run: true,
+            },
+        },
+    },
+    {
+        files: [
+            'packages/@lwc/integration-tests/src/**/!(*.spec.js)',
+            'packages/@lwc/integration-karma/test/**',
+            'packages/@lwc/integration-karma/test-hydration/**',
+        ],
+
+        rules: {
+            'header/header': 'off',
+        },
+    },
+    {
+        files: ['packages/@lwc/integration-karma/**'],
+
+        languageOptions: {
+            globals: {
+                lwcRuntimeFlags: true,
+            },
+        },
+    },
+    ...compat.extends('plugin:@typescript-eslint/disable-type-checked').map((config) => ({
+        ...config,
+        files: ['**/.only'],
+    })),
+    {
+        files: ['**/.only'],
+
+        rules: {
+            'header/header': 'off',
+            '@lwc/lwc-internal/forbidden-filename': 'error',
+        },
+    },
+    ...compat.extends('plugin:@typescript-eslint/disable-type-checked').map((config) => ({
+        ...config,
+        files: ['**/.skip'],
+    })),
+    {
+        files: ['**/.skip'],
+
+        rules: {
+            'header/header': 'off',
+            '@lwc/lwc-internal/forbidden-filename': 'off',
+        },
+    },
```

Comment:
```suggestion
    {
        // These are empty files used to help debug test fixtures
        files: ['**/.only', '**/.skip'],
        plugins: { '@lwc/lwc-internal': lwcInternal },
        rules: {
            '@lwc/lwc-internal/forbidden-filename': 'error',
        },
    },
```
`.only` and `.skip` aren't recognized by eslint, so I think all we need to do is enable the one rule that we actually want to enforce on them. (Not sure whether the `plugins` line is needed or covered by another section - included it to be safe.)